src/metabase/driver/sql_jdbc/connection.clj

```diff
@@ -103,6 +105,20 @@ For setting the maximum, see [MB_APPLICATION_DB_MAX_CONNECTION_POOL_SIZE](#mb_ap
                     (long (/ qp.pipeline/*query-timeout-ms* 1000))))
   :setter     :none)
 
+(setting/defsetting jdbc-data-warehouse-debug-unreturned-connection-stack-traces
+  "Tell c3p0 to log a stack trace for any connections killed due to exceeding the timeout specified in
+  [[jdbc-data-warehouse-unreturned-connection-timeout-seconds]].
+
+  Note: even if this value is set to true, the exceptions are not logged by default because we set the com.mchange log
+  level to ERROR in our log4j2.xml config, and c3p0 logs the exceptions at INFO level. You need to update the
+  com.mchange log level to INFO via a custom log4j config in order to see the stack traces in the logs."
+  :visibility :internal
+  :type       :boolean
+  :default    false
+  :export?    false
+  :setter     :none
+  :doc        false)
```

Comment:
Do we want to document this? I had assumed not, since I thought we also didn't document the closely-related `jdbc-data-warehouse-unreturned-connection-timeout-seconds`, above (not .

But now I see that `MB_JDBC_DATA_WAREHOUSE_UNRETURNED_CONNECTION_TIMEOUT_SECONDS` *is* documented in a pair of markdown files

```
File: docs/configuring-metabase/environment-variables.md
1870:6:### `MB_JDBC_DATA_WAREHOUSE_UNRETURNED_CONNECTION_TIMEOUT_SECONDS`

File: src/metabase/cmd/resources/other-env-vars.md
212:6:### `MB_JDBC_DATA_WAREHOUSE_UNRETURNED_CONNECTION_TIMEOUT_SECONDS`
```

which then apparently gets included in the docs [here](https://www.metabase.com/docs/latest/configuring-metabase/environment-variables#mb_jdbc_data_warehouse_max_connection_pool_size), so maybe we want a similar doc entry for this new setting?

Reply:
There seems to be no connection between the `:doc` property and the documentation in the markdown files.

Reply:
I think `src/metabase/cmd/env_var_dox.clj` generates `docs/configuring-metabase/environment-variables.md` and will use `:doc` if it finds it and will also just include the stuff in `other-env-vars.md`, but it's not clear to me when something should be doc'ed via `:doc` vs manually in `other-env-vars.md` vs not documented at all.

At the top of of `other-env-vars.md` I find

> The following environment variables can only be set via the environment. They cannot be set by the configuration file.

but I'm not sure what determines if a setting can be set via the config file or not. Perhaps some combination of `:visibility :internal` and/or `:setter :none`?

Reply:
The function that includes `other-env-vars.md` in the `env_var_dox` cmd says that the file contains docs for env vars not defined by `defsetting`

```clojure
(defn- non-defsetting-env-vars
  "Retrieves environment variables not specified via `defsetting`."
  []
  (str "\n\n" (slurp "src/metabase/cmd/resources/other-env-vars.md") "\n"))
```

but `MB_JDBC_DATA_WAREHOUSE_UNRETURNED_CONNECTION_TIMEOUT_SECONDS` is documented in that file and yet is defined by a `defsetting` here

https://github.com/metabase/metabase/blob/dba17e6c3699b37da8bcb747688a9f5f24f74ff7/src/metabase/driver/sql_jdbc/connection.clj#L94

It seems that anything with `:setter :none` or `:doc :false` is intentionally excluded from doc generation, so maybe the name/docstring of `non-defsetting-env-vars` is just out of date. Maybe the file used to be for non-defsetting vars and over time expanded to include some defsettings that are nevertheless env-var-only.